internal/compliance/snapshot_poller/models/aws/types.go

```diff
@@ -64,12 +67,63 @@ type GenericAWSResource struct {
 
 // ResourcePollerInput contains the metadata to request AWS resource info.
 type ResourcePollerInput struct {
-	AuthSource          *string
-	AuthSourceParsedARN arn.ARN
-	IntegrationID       *string
-	Region              *string
-	Timestamp           *time.Time
-	NextPageToken       *string
+	AuthSource              *string
+	AuthSourceParsedARN     arn.ARN
+	IntegrationID           *string
+	Region                  *string
+	Timestamp               *time.Time
+	NextPageToken           *string
+	RegionIgnoreList        []string
+	ResourceTypeIgnoreList  []string
+	ResourceRegexIgnoreList []string
+	CompiledRegexIgnoreList []*regexp.Regexp
+}
+
+func (r *ResourcePollerInput) CompileRegex() error {
+	r.CompiledRegexIgnoreList = make([]*regexp.Regexp, len(r.ResourceRegexIgnoreList))
+
+	for i, glob := range r.ResourceRegexIgnoreList {
+		if glob == "" {
+			r.CompiledRegexIgnoreList[i] = nil
+		}
+		// First,  escape any regex special characters
+		escaped := regexp.QuoteMeta(glob)
+
+		// Wildcards in the original pattern are now escaped literals - convert back
+		// NOTE: currently no way for user to specify a glob that would match a literal '*'
+		regex := "^" + strings.ReplaceAll(escaped, `\*`, `.*`) + "$"
+		compiledGlob, err := regexp.Compile(regex)
+		if err != nil {
+			// We are building the regex, so it should always be valid
+			zap.L().Error("invalid regex",
+				zap.String("originalPattern", glob),
+				zap.String("transformedRegex", regex),
+				zap.Error(err),
+			)
+			return err
+		}
+		r.CompiledRegexIgnoreList[i] = compiledGlob
+	}
+	return nil
+}
+
+func (r *ResourcePollerInput) ShouldIgnoreResource(resourceID string) (ignore bool, err error) {
+	// Check if the regexs have been compiled
+	if r.CompiledRegexIgnoreList == nil {
+		err := r.CompileRegex()
+		if err != nil {
+			return false, err
+		}
+	}
+	for _, compiledRegex := range r.CompiledRegexIgnoreList {
+		if compiledRegex == nil {
+			continue
+		}
+		if compiledRegex.MatchString(resourceID) {
+			return true, nil
+		}
+	}
```

Comment:
With the above suggestion, you don't have to worry about `nil` entries.

```suggestion
	for _, compiledRegex := range r.CompiledRegexIgnoreList {
		if compiledRegex.MatchString(resourceID) {
			return true, nil
		}
	}
```